src/cmd/troubleshoot/dynakube.go

```diff
@@ -0,0 +1,280 @@
+		return err
+	}
+
+	_, hasConfig := secret.Data[pullSecretFieldName]
+	if !hasConfig {
+		logErrorf("token '%s' does not exist in '%s:%s' secret", pullSecretFieldName, troubleshootCtx.namespaceName, troubleshootCtx.customPullSecretName)
+		return fmt.Errorf("token '%s' does not exist in '%s:%s' secret", pullSecretFieldName, troubleshootCtx.namespaceName, troubleshootCtx.customPullSecretName)
+	}
+
+	logInfof("secret token '%s' exists", pullSecretFieldName)
+	return nil
+}
+
+func evaluateProxySecret(troubleshootCtx *troubleshootContext) error {
+	query := kubeobjects.NewDynakubeQuery(nil, troubleshootCtx.apiReader, troubleshootCtx.namespaceName).WithContext(context.TODO())
+	dynakube, err := query.Get(types.NamespacedName{Namespace: troubleshootCtx.namespaceName, Name: troubleshootCtx.dynakubeName})
+	if err != nil {
+		logWithErrorf(err, "selected '%s:%s' Dynakube does not exist", troubleshootCtx.namespaceName, troubleshootCtx.dynakubeName)
+		return err
+	}
+
+	if dynakube.Spec.Proxy == nil || dynakube.Spec.Proxy.ValueFrom == "" {
+		logInfof("proxy secret not used")
+		return nil
+	}
+
+	troubleshootCtx.proxySecretName = dynakube.Spec.Proxy.ValueFrom
+	logInfof("'%s:%s' proxy secret is used", troubleshootCtx.namespaceName, troubleshootCtx.proxySecretName)
+	return nil
+}
+
+func checkIfProxySecretWithTheGivenNameExists(troubleshootCtx *troubleshootContext) error {
+	if troubleshootCtx.proxySecretName == "" {
+		return nil
+	}
+
+	query := kubeobjects.NewSecretQuery(context.TODO(), nil, troubleshootCtx.apiReader, log)
+	if _, err := query.Get(types.NamespacedName{Namespace: troubleshootCtx.namespaceName, Name: troubleshootCtx.proxySecretName}); err != nil {
+		logWithErrorf(err, "'%s:%s' proxy secret is missing", troubleshootCtx.namespaceName, troubleshootCtx.proxySecretName)
+		return err
+	}
+
+	logInfof("custom pull secret '%s:%s' exists", troubleshootCtx.namespaceName, troubleshootCtx.proxySecretName)
+	return nil
+}
+
+func checkProxySecretHasRequiredTokens(troubleshootCtx *troubleshootContext) error {
+	if troubleshootCtx.proxySecretName == "" {
+		return nil
+	}
+
+	query := kubeobjects.NewSecretQuery(context.TODO(), nil, troubleshootCtx.apiReader, log)
+	secret, err := query.Get(types.NamespacedName{Namespace: troubleshootCtx.namespaceName, Name: troubleshootCtx.proxySecretName})
+	if err != nil {
+		logWithErrorf(err, "'%s:%s' proxy secret is missing", troubleshootCtx.namespaceName, troubleshootCtx.proxySecretName)
+		return err
+	}
+
+	_, hasProxy := secret.Data[dtclient.CustomProxySecretKey]
+	if !hasProxy {
+		logErrorf("token '%s' does not exist in '%s:%s' secret", dtclient.CustomProxySecretKey, troubleshootCtx.namespaceName, troubleshootCtx.proxySecretName)
+		return fmt.Errorf("token '%s' does not exist in '%s:%s' secret", dtclient.CustomProxySecretKey, troubleshootCtx.namespaceName, troubleshootCtx.proxySecretName)
```

Comment:
why are we logging this and returning the error?

Reply:
Fixed - an error is returned only and printed to log by the caller